src/connectors/SingularityConnector.ts

```diff
@@ -195,6 +251,19 @@ class SingularityConnector extends SlurmConnector {
 
     return jobENV;
   }
+
+  /**
+   * Creates a bash script using kernelConfig
+   * @param{executableManifest} manifest - manifest that needs toe be executed
+   */
+  async createBashScript(manifest: executableManifest){
```

Comment:
Maybe let's rename this function to "createKernelInit" because "createBashScript" is overly generic and doesn't really describe well what it is doing.